From 644fbddd9209404a69c0f248e2f5f62e999799d6 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Tue, 15 Oct 2024 16:58:13 +0300
Subject: [PATCH] Replace LT_ARGZ_H_PATH usage with bool for LTARGZH_EXISTS

* m4/ltdl.mk: Add conditional with LTARGZH_EXISTS to append path for
  LT_ARGZ_H file to BUILT_SOURCES.
* m4/ltargz.m4: Remove LT_ARGZ_H_PATH.
* m4/ltdl.m4: Add AM_CONDITIONAL for LTARGZH_EXISTS that is set to
  true if LT_ARGZ_H is non-empty.
---
 libltdl/ltdl.mk | 4 +++-
 m4/ltargz.m4    | 2 --
 m4/ltdl.m4      | 3 ++-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/libltdl/ltdl.mk b/libltdl/ltdl.mk
index 2e224cbc..9d6be829 100644
--- a/libltdl/ltdl.mk
+++ b/libltdl/ltdl.mk
@@ -131,7 +131,9 @@ EXTRA_DIST	       += libltdl/COPYING.LIB \
 ## Gnulib Makefile.am snippets ##
 ## --------------------------- ##
 
-BUILT_SOURCES	+= $(LT_ARGZ_H_PATH)
+if LTARGZH_EXISTS
+BUILT_SOURCES	+= libltdl/libltdl/$(LT_ARGZ_H)
+endif
 EXTRA_DIST	+= libltdl/libltdl/lt__argz_.h \
 		   libltdl/lt__argz.c
 
diff --git a/m4/ltargz.m4 b/m4/ltargz.m4
index 327b0b47..7635ce8e 100644
--- a/m4/ltargz.m4
+++ b/m4/ltargz.m4
@@ -71,9 +71,7 @@ AS_IF([test -z "$LT_ARGZ_H"],
         [AC_DEFINE([HAVE_WORKING_ARGZ], 1,
                    [This value is set to 1 to indicate that the system argz facility works])],
         [LT_ARGZ_H=lt__argz.h
-        LT_ARGZ_H_PATH=libltdl/libltdl/$LT_ARGZ_H
         AC_LIBOBJ([lt__argz])])])
 
 AC_SUBST([LT_ARGZ_H])
-AC_SUBST([LT_ARGZ_H_PATH])
 ])
diff --git a/m4/ltdl.m4 b/m4/ltdl.m4
index c47bb98c..22761e6e 100644
--- a/m4/ltdl.m4
+++ b/m4/ltdl.m4
@@ -435,7 +435,8 @@ esac
 
 m4_ifdef([AM_CONDITIONAL],
 [AM_CONDITIONAL(INSTALL_LTDL, test no != "${enable_ltdl_install-no}")
- AM_CONDITIONAL(CONVENIENCE_LTDL, test no != "${enable_ltdl_convenience-no}")])
+ AM_CONDITIONAL(CONVENIENCE_LTDL, test no != "${enable_ltdl_convenience-no}")
+ AM_CONDITIONAL(LTARGZH_EXISTS, test -n "$LT_ARGZ_H")])
 ])# _LT_ENABLE_INSTALL
 
 
-- 
2.45.2

